### PR TITLE
Split RabbitMQPlugin into ExtractorRoutingService and MessageService

### PR DESCRIPTION
**Pull request by max-zilla**
This attempts to move the logic to route RMQ messages based on file mime type and other events, from the actual code to connect to RMQ and send/fetch messages. One impact is that this would make implementing alternative message bus system easier.

Also integrates @bodom0015 's new EventSinkService that pushes events to a standard queue for ingest to reporting databases.

### CODE REVIEW

**Comment by lmarini** [comment]
@max-zilla @bodom0015 I am getting a client side error when uploading a file on the file upload page. The file uploads fine to the dataset, but the filed entry disappears from the list after upload on the file upload page. Can you try reproducing it?

The console errors is:

`dataset-attach-fileuploader.js:20 Uncaught TypeError: Cannot read property 'split' of undefined
    at dataset-attach-fileuploader.js:20`

```
var rowlist = document.getElementById("filelist_table").rows;
			for (var row of rowlist) {
				var fileid = row.innerHTML.split('data-url="')[1].split('">')[0]
				fileid = fileid.substring(fileid.indexOf("/api/files/")+"/api/files/".length).replace("/remove","");
				fileids.push(fileid);
			}
```

**Review by bodom0015 — APPROVED** [review]
LGTM :+1:

Added a couple of commits to fix the failing test case(s) and to parameterize `exchange_type` in `MessageService.submit()`

**Review by lmarini — CHANGES_REQUESTED** [review]

**Comment by lmarini** [comment]
> @max-zilla @bodom0015 I am getting a client side error when uploading a file on the file upload page. The file uploads fine to the dataset, but the filed entry disappears from the list after upload on the file upload page. Can you try reproducing it?
> 
> The console errors is:
> 
> `dataset-attach-fileuploader.js:20 Uncaught TypeError: Cannot read property 'split' of undefined at dataset-attach-fileuploader.js:20`
> 
> ```
> var rowlist = document.getElementById("filelist_table").rows;
> 			for (var row of rowlist) {
> 				var fileid = row.innerHTML.split('data-url="')[1].split('">')[0]
> 				fileid = fileid.substring(fileid.indexOf("/api/files/")+"/api/files/".length).replace("/remove","");
> 				fileids.push(fileid);
> 			}
> ```

pushed a fix

File: app/api/Status.scala [hunk 1]
```diff
@@ -69,22 +69,6 @@ class Status @Inject()(spaces: SpaceService,
...
       }
 
-      // rabbitmq
-      case p: RabbitmqPlugin => {
-        val status = if (p.connect) {
-          "connected"
-        } else {
-          "disconnected"
-        }
-        result.put("rabbitmq", if (Permission.checkServerAdmin(user)) {
```
[the remark this answers is not in the document]
Comment (lmarini) [reply app/api/Status.scala:79]: Pushed a fix that puts this back in. Created a getInfo() function to let the implementation specify what to use. 
Might want to use the same pattern for future plugin -> services conversions.

**Review by lmarini — APPROVED** [review]

**Pull request merged**